kotlin_money/src/main/java/com/github/paylike/kotlin_money/PaymentAmount.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.paylike.kotlin_money
+
+import com.github.paylike.kotlin_currencies.PaylikeCurrency
+
+
+class PaymentAmount(_currency: PaylikeCurrency, _value: Long, _exponent: Int) {
+    val currency: PaylikeCurrency = _currency
+    val value: Long = _value
+    val exponent: Int = _exponent
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        other as PaymentAmount
+        if (this.currency.code != other.currency.code) return false
+        return this.value == other.value && this.exponent == other.exponent
+    }
+
+    override fun hashCode(): Int = currency.hashCode() + value.hashCode() + exponent.hashCode()
```

Comment:
Do we need this?

Reply:
The hashCode? If we override the equals then we shall override the hashCode as well.
https://www.geeksforgeeks.org/override-equalsobject-hashcode-method/